File: src/Module/Role/RoleEntity.java
package Module.Role;

import Module.Permission.PermissionEntity;
import Module.Permission.PermissionModel;

import java.io.Serializable;
import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class RoleEntity implements Serializable {
    public int id;
    public String name;
    public String description;
    public List<PermissionEntity> permissionEntities;

    public RoleEntity() {
    }

    public RoleEntity(RoleModel RoleModel, Object... objects) {
        this.id = RoleModel.getId();
        this.name = RoleModel.getName();
        this.description = RoleModel.getDescription();
        for (Object object : objects) {
            if (object instanceof Collection) {
                for (Object o : (Collection<Object>) object) {
                    if (o instanceof PermissionModel) {
                        this.permissionEntities = ((Collection<PermissionModel>) object).parallelStream().map(PermissionEntity::new).collect(Collectors.toList());
                        break;
                    }
                }
            }
        }
    }

    public RoleModel toModel() {
        RoleModel RoleModel = new RoleModel();
        RoleModel.setId(id);
        RoleModel.setDescription(description);
        RoleModel.setName(name);
        return RoleModel;
    }
}

File: web/Angular4/src/app/Modules/Role/Role.Entity.ts
import {PermissionEntity} from "../Permission/Permission.Entity";

export class RoleEntity {

    id: number;
    name: string;
    description : string;
    permissionEntities : PermissionEntity[];


    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;

    constructor(data: any = null) {
        if (data == null) {
            this.id = null;
            this.name = null;
            this.description = null;
            this.permissionEntities = [];
        } else {
            this.id = data.id;
            this.name = data.name;
            this.description = data.description;
            if (data.permissionEntities != null) {
                this.permissionEntities = [];
                for (let item of data.permissionEntities) {
                    this.permissionEntities.push(new PermissionEntity(item));
                }
            }
            if (this.permissionEntities == null) this.permissionEntities = [];
        }
        this.IsEdit = false;
    }
}